[DX-492] Added worked example for OPA Additional Permissions

rewsmith opened the pull request:
### For internal users - please add a Jira DX PR ticket to the subject!
https://tyktech.atlassian.net/browse/DX-492

---
### Preview Link
<!-- Add a direct preview link to make it easier for the reviewer to review. Best to add a few direct links to the pages in the PR -->
https://deploy-preview-3120--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-dashboard/advanced-configurations/open-policy-agent/opa-permissions-example/

### Description
Following feedback from a customer that they were unable to understand how to implement an OPA rule using Additional Permissions, this change provides a worked end-to-end example.

#### Screenshots (if appropriate)

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply -->

- [ x] I have added a **preview link** to the PR description.
- [ x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ x] Make sure you have started *your change* off *our latest `master`*.
- [ x] I **labelled** the PR
<!-- Label your PR according to the type of changes that your code introduces. This ensures that we know how/when to publish the PR. These are the options:
- Fixing typo (please merge to production) - add the label `now`
- Documenting a new feature (please merge to production) - add the label `now`
- Documentation for future release (please do not merge to production) - add label `future release` and label of the release
- [Something else (please add if needs merging to production or not) - add label according to the above logic -->

dcs3spp reviewed (changes requested):
Hi @rewsmith many thanks for contributing the worked example for OPA permissions. 

The PR looks good to me. I have had a read through and made some minor suggestions.

Please take a look at the [New Guide](https://tyktech.atlassian.net/wiki/spaces/EN/pages/2118287411/Creating+Content+For+The+New+Tyk+Docs+Website) for creating PR content on the new website.

For PRs that include new content files there is an additional step for CI checks to pass.

Since the new website has been released there is now a menu.yaml file that maps the menu structure. There are new fixed folders in _tyk-docs/content_ folder added for the purpose of storing fresh content. These folders are explained in the above guide and correspond to the top nav bar navigation tabs. The purpose of each tab is explained [here](https://tyktech.atlassian.net/wiki/spaces/EN/pages/2120187942/New+Docs+Information+Architecture+and+Content+Sections)

The new content file in the PR needs to be placed in one of the new folders, maybe in _product stack -> dashboard -> troubleshooting section_ or _developer support -> frequently asked questions?_

Then it needs to be referenced in the menu.yaml file. The guide gives an example [here](https://tyktech.atlassian.net/wiki/spaces/EN/pages/2118287411/Creating+Content+For+The+New+Tyk+Docs+Website#Adding-A-New-Page)

Once changes have been made let me know if you need any help with the menu.yaml aspect

rewsmith commented on tyk-docs/content/tyk-dashboard/opa-permissions-example.md:
```diff
@@ -0,0 +1,255 @@
+
+#### Sample Request
+
+In order to add the new role/permissions use the following payload.  Remember to set the `authorization` header to your Tyk Dashboard API Access Credentials secret from your user profile on the Dashboard UI
```
Makes sense yes, I'll make the change

rewsmith commented on tyk-docs/content/tyk-dashboard/opa-permissions-example.md:
```diff
@@ -0,0 +1,255 @@
+{{< note success >}}
+**Note**  
+
+This assumes no other additional permissions already exist.  If you're adding to existing permissions you'll want to send a GET to `/api/org/permissions` first, and then add the new permission to the existing list.
```
A PUT request will replace the entire permissions list yes, and that's why I'm suggesting a GET request first such that any existing permissions can be incorporated.  Basically trying to protect against any existing permissions being overwritten/lost

rewsmith commented on tyk-docs/content/tyk-dashboard/opa-permissions-example.md:
```diff
@@ -0,0 +1,255 @@
+}
+# Do not allow users (excluding admin users) to reset the password of another user.
+deny[x] {
+	request_permission[_] = "ResetPassword"
```
ResetPassword is only included as part of an example full updated rule set and isn't part of the change itself.  I included the example complete updated rule set to hopefully make the required change clearer.  That set, if the default rule set changes in future, this example will be out of date

rewsmith commented:
@dcs3spp As well as accepting your changes I've moved the content file to the same directory as the other OPA files under /product-stack/tyk-dashboard/advanced-configurations/open-policy-agent, and added a corresponding entry into menu.yaml.  There's a couple of comments in the conversation to agree on, and then hopefully we're good to merge.

dcs3spp commented:
> https://deploy-preview-3116--tyk-docs.netlify.app/docs/tyk-dashboard/opa-permissions-example

Thanks @rewsmith Almost there, when I try and use the preview link to navigate to the example I cannot see it in the OPA examples in the menu

https://deploy-preview-3116--tyk-docs.netlify.app/docs/nightly/

dcs3spp commented:
@rewsmith I have approved. If we could get a peer review accepted then I think we are ready to merge this

dcs3spp commented:
/release to release-5.1
